public/scripts/upload.js

```diff
@@ -482,6 +492,12 @@ function updateFileListItem(itemNode, imageUpload) {
     } else {
       previewNode.classList.add("file-list-item__preview_non_selected");
     }
+
+    previewNode.onclick = handlePreviewImage;
+    fileInfoNode.onclick = handlePreviewImage;
```

Comment:
prefer addEvenetListener due to [this](https://stackoverflow.com/a/6348533/19287186)

Reply:
@pkong-ds using addEventListener, it will make the `fileInfoNode` clickable and scrollable to the preview area when generating a preview, because after uploading the image the image will be in a successful state but has not started generating previews yet, how about add a state `readyForPreview` after uploading image, and only set success state after generating preview?

Reply:
Sorry I don't understand. How does `onclick`  not have the same problem?

Reply:
there is how the state of `imageUpload` change:

`ReadyForRead` -> `Reading` -> `ReadSuccess` -> `GeneratingPreview` -> `ReadSuccess`
so it will call `addEventListener` before generating preview, which means user can click the item before generating preview

ref: `read()` in `public/scripts/models/image-upload.js`

Reply:
> ReadyForRead -> Reading -> ReadSuccess -> GeneratingPreview -> ReadSuccess
so it will call addEventListener before generating preview, which means user can click the item before generating preview

Thanks, I understood this part

What I do not understand is, if `xxx.addEvenetListener('click', cb)` has a problem of 

> make the fileInfoNode clickable and scrollable to the preview area when generating a preview

Why does `xxx.onclick = cb` not have the same problem?

Reply:
as per offline discussion, so now it became

ReadyForRead -> Reading -> ReadyForPreview -> GeneratingPreview -> ReadSuccess

Reply:
sitenote: Maybe we should use a state machine for this :sadfrog: so hard to keep track